src/group_impl.h

```diff
@@ -231,6 +235,7 @@ static void secp256k1_ge_table_set_globalz(size_t len, secp256k1_ge *a, const se
         secp256k1_fe_verify(&zr[i]);
         /* Ensure all y values are in weak normal form for fast negation of points */
         secp256k1_fe_normalize_weak(&a[i].y);
+        secp256k1_ge_verify(&a[i]);
```

Comment:
nit: I think the functions would be easier to read, if they all follow the same pattern (in particular the functions with loops:)
 * on entry, verify all input ge, gej
 * (possibly an empty line if you're willing to add this)
 * actual function body
 * (possibly an empty line if you're willing to add this)
 * on exit, verify all output gej, possibly separated by a newline

Reply:
Started working on that but got doubt, as I think that would need additional verification _loops_ at the start and end of a function if the input is an array of ge(j) instances (like in secp256k1_ge_table_set_globalz). Is that what you meant? (If yes, maybe good follow-up material?)

Reply:
> Is that what you meant?

Yes, this would decouple the verification from the actual code. (And loops could be in `#ifdef VERIFY`, though any sane compiler would remove the empty loop.)

> (If yes, maybe good follow-up material?)

Okay, fair. I'm happy with any way in the end.

My thinking was that the advantage of doing it in this PR is that it will be easier to check for reviewers that verification is present everywhere. (The diff of that commit will be large, but noone will need to read the diff; they can just check if the result catches everything.) But I'm convinced of this already because I did the work to go through the `group_impl.h`.

If you think a separate PR for this refactor is better, then yet another way to do this would be to have a separate PR *before this PR*. But this may require more rebasing work.

I believe you can decide best now that you looked into this.

Reply:
> My thinking was that the advantage of doing it in this PR is that it will be easier to check for reviewers that verification is present everywhere. (The diff of that commit will be large, but noone will need to read the diff; they can just check if the result catches everything.)

That makes sense! Thanks for elaborating, the motivation is now clear to me. Added the suggestion in the first commit. Should be quite trivial to review, only the changes in `secp256k1_ge_table_set_globalz` are a bit more invasive, as the initialization of `i` for the backwards loop is now done later, due to `i` also being used for the verification loop of inputs before. But I think that's actually even a slight improvement because we avoid a potential integer underflow for `i` in the case of `len == 0` (even though that's more theoretical, in practice no-one would call that function with a zero-length I guess).

I decided to put the input/output verification loops in `#ifdef VERIFY`, mostly for visual reasons (I'd also assume that even without those, most compilers would optimize the loops away anyway), but also happy to remove them if wished.